code/modules/ai/ai_behaviors/ai_behavior.dm

```diff
@@ -6,74 +6,125 @@ Registers signals, handles the pathfinding element addition/removal alongside ma
 */
 
 /datum/ai_behavior
+	///What atom is the ai moving to
+	var/atom/atom_to_walk_to
+	///How far should we stay away from atom_to_walk_to
+	var/distance_to_maintain = 1
+	///Prob chance of sidestepping (left or right) when distance maintained with target
+	var/sidestep_prob = 0
+	///Current node to use for calculating action states: this is the mob's node
+	var/obj/effect/ai_node/current_node
+	///What the ai is doing right now
+	var/current_action
+	///The standard ation of the AI, aka what it should do at the init or when going back to "normal" behavior
+	var/base_action = MOVING_TO_NODE
+	///Ref to the parent associated with this mind
+	var/mob/mob_parent
+	///An identifier associated with this behavior, used for accessing specific values of a node's weights
+	var/identifier
+	///How far will we look for targets
+	var/target_distance = 8
+	///What we will escort
+	var/atom/escorted_atom
+	///When this timer is up, we force a change of node to ensure that the ai will never stay stuck trying to go to a specific node
+	var/anti_stuck_timer
```

Comment:
since we are not doing a smart pathfinder, sometimes we are finding another node to go to.

With this, it's impossible for xeno to stay stuck